From c8c4aac6d68bc864fa197d0bef0a747c70dd67f3 Mon Sep 17 00:00:00 2001
From: Olivier Fourdan <ofourdan@redhat.com>
Date: Thu, 12 Dec 2024 14:05:04 +0100
Subject: [PATCH] search-app: Use firefox from flatpak

Firefox is being installed through flatpak these days, adjust the search
appliance to use flatpak instead of invoking firefox directly.

https://issues.redhat.com/browse/RHEL-36521
---
 search-app/org.gnome.Kiosk.SearchApp.desktop.in.in      | 2 +-
 search-app/systemd/org.gnome.Kiosk.SearchApp.service.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/search-app/org.gnome.Kiosk.SearchApp.desktop.in.in b/search-app/org.gnome.Kiosk.SearchApp.desktop.in.in
index 478b328..ad142b8 100644
--- a/search-app/org.gnome.Kiosk.SearchApp.desktop.in.in
+++ b/search-app/org.gnome.Kiosk.SearchApp.desktop.in.in
@@ -2,7 +2,7 @@
 Type=Application
 Name=Search Appliance
 Comment=Sample Search Appliance Application for GNOME Kiosk
-Exec=@bindir@/firefox --kiosk --private-window --new-instance  https://www.google.com
+Exec=@bindir@/flatpak run --command=firefox --file-forwarding org.mozilla.firefox --kiosk --private-window --new-instance https://www.google.com
 Categories=Core;System;
 NoDisplay=true
 X-GNOME-HiddenUnderSystemd=true
diff --git a/search-app/systemd/org.gnome.Kiosk.SearchApp.service.in b/search-app/systemd/org.gnome.Kiosk.SearchApp.service.in
index 29ddcd8..d4f5d0f 100644
--- a/search-app/systemd/org.gnome.Kiosk.SearchApp.service.in
+++ b/search-app/systemd/org.gnome.Kiosk.SearchApp.service.in
@@ -4,5 +4,5 @@ BindsTo=gnome-session.target
 After=gnome-session.target
 
 [Service]
-ExecStart=@bindir@/firefox --kiosk --private-window --new-instance  https://www.google.com
+ExecStart=@bindir@/flatpak run --command=firefox --file-forwarding org.mozilla.firefox --kiosk --private-window --new-instance https://www.google.com
 Restart=always
-- 
2.51.1

